app/coffee/RedisManager.coffee

```diff
@@ -113,7 +113,17 @@ module.exports = RedisManager =
 			if doc_project_id? and doc_project_id isnt project_id
 				logger.error project_id: project_id, doc_id: doc_id, doc_project_id: doc_project_id, "doc not in project"
 				return callback(new Errors.NotFoundError("document not found"))
-			callback null, docLines, version, ranges
+
+			# doc is not in redis, bail out
+			if !lines?
```

Comment:
If so, probably a sign that this code pathway split needs a unit test to test each branch